### PR TITLE
lein: add page

### PR DESCRIPTION
**Pull request by OrBaruk**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/lein.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# lein
+
+> Manage clojure projects with declarative configuration.
+
+- Generate scaffolding for a new project based on a template:
+
+`lein new {{template}} {{project_name}}`
```
Comment (sbrl) [inline pages/common/lein.md]: Is this the path to the template or the template name? Perhaps we could change this to `{{template_name}}` or `{{path/to/template}}` as appropriate.

File: pages/common/lein.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- Start a repl session either with the project or standalone:
+
+`lein repl`
+
+- Run the project's -main function with optional args:
```
Comment (sbrl) [inline pages/common/lein.md]: Is the `-` a typo here? Also, I think `main` may want to be enclosed in backticks.

File: pages/common/lein.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+`lein run {{args}}`
+
+- Run the project's tests:
+
+`lein test`
+
+- Package up the project files and all dependencies into a jar file:
```
Comment (sbrl) [inline pages/common/lein.md]: A minor readability tweak, but perhaps `and all its dependencies` would sound better?

File: pages/common/lein.md (hunk 2 above)
Comment (OrBaruk) [reply pages/common/lein.md]: No, it is actually called `-main` function, I wasn't sure if backticks where used outside the command example.

**Comment by sbrl** [comment]
Erm oops! Didn't mean to merge this before @agnivade / @waldyrious reviewed :P Looks goo to me though.

**Pull request merged**